src/core/server/server.api.md

```diff
@@ -1110,14 +1111,17 @@ export type RouteContentType = 'application/json' | 'application/*+json' | 'appl
 // @public
 export type RouteMethod = 'get' | 'post' | 'put' | 'delete' | 'patch' | 'options';
 
+// @public
+export type RouterHandler<Method extends RouteMethod> = <P extends ObjectType, Q extends ObjectType, B extends ObjectType | Type<Buffer> | Type<Stream>>(route: RouteConfig<P, Q, B, Method>, handler: RequestHandler<P, Q, B, Method>) => void;

```

Comment:
different type order in generic could create confusion for plugin authors. https://github.com/elastic/kibana/pull/50783/files/93d267c3910c5ccebfa10e0ec0600c7abecbe22e..7d4afae6a283634d65f6987f26ad340d946822ca?file-filters%5B%5D=.md&file-filters%5B%5D=.ts&hide-deleted-files=true#diff-f9b3398cad5ffd2f60f87bc0c65e1fedR58
we will have to create lots of `RouterHandler` and `KibanaRequest` manually during migration

Reply:
Moved both to the end so we maintain the order.